R/fullVALIDATIONREMIND.R

```diff
@@ -17,42 +17,65 @@ fullVALIDATIONREMIND <- function(rev = 0) {
   for (mapping in c(getConfig("regionmapping"), getConfig("extramappings"))) {
-    columns <- setdiff(colnames(toolGetMapping(mapping, 'regional')),
-                       c('X', 'CountryCode'))
+    columns <- setdiff(colnames(toolGetMapping(mapping, "regional")),
+                       c("X", "CountryCode"))
 
-    if (any(columns %in% rel))
-      warning("The following column(s) from ", mapping,
-              " exist in another mapping an will be ignored: ",
-              paste(columns[columns %in% rel], collapse = ", "))
+    if (any(columns %in% rel)) {
+      warning(
+        "The following column(s) from ", mapping,
+        " exist in another mapping an will be ignored: ",
+        paste(columns[columns %in% rel], collapse = ", ")
+      )
+    }
 
     rel <- unique(c(rel, columns))
   }
 
-  columnsForAggregation <- gsub("RegionCode", "region",
-                                paste(rel, collapse = "+"))
+  columnsForAggregation <- gsub(
+    "RegionCode", "region",
+    paste(rel, collapse = "+")
+  )
 
   # historical data ----
   valfile <- "historical.mif"
 
-  calcOutput("Historical", round = 5,  file = valfile, aggregate = columnsForAggregation,
-             append = FALSE, warnNA = FALSE, try = TRUE)
+  calcOutput("Historical",
+    round = 5, file = valfile, aggregate = columnsForAggregation,
+    append = FALSE, warnNA = FALSE, try = TRUE
+  )
 
-  calcOutput(type = "IEA_ETP", aggregate = "global", file = valfile,
-             append = TRUE, warnNA = FALSE, try = TRUE, isValidation = TRUE)
+  calcOutput(
+    type = "IEA_ETP", aggregate = columnsForAggregation, file = valfile,
+    append = TRUE, warnNA = FALSE, try = TRUE, isValidation = TRUE
+  )
 
-  calcOutput(type = "IEA_ETP", aggregate = "region", file = valfile,
-             append = TRUE, warnNA = FALSE, try = TRUE, isValidation = TRUE)
+  ## industry value added ----
+  calcOutput(
+    type = "UNIDO", subtype = "INDSTAT2", file = valfile,
+    aggregate = columnsForAggregation, append = TRUE, warnNA = FALSE,
+    try = TRUE
+  )
 
-  calcOutput(type = "IEA_WEO_2021", aggregate = "global", file = valfile,
-             append = TRUE, warnNA = FALSE, try = TRUE, isValidation = TRUE)
+  ## add WEO data on regional and global level ----
+  weo <- calcOutput(
+    type = "IEA_WEO_2021", subtype = "global", aggregate = columnsForAggregation,
+    warnNA = FALSE, try = TRUE, isValidation = TRUE
+  )
 
-  calcOutput(type = "IEA_WEO_2021", aggregate = "region", file = valfile,
-             append = TRUE, warnNA = FALSE, try = TRUE, isValidation = TRUE)
+  weo <- weo["GLO", , ]
+  write.report(weo, file = file.path(getConfig("outputfolder"), valfile), append = TRUE)
+
+  weo <- calcOutput(
+    type = "IEA_WEO_2021", subtype = "region", aggregate = columnsForAggregation,
+    warnNA = FALSE, try = TRUE, isValidation = TRUE
+  )
+
+  weo <- weo["GLO", , invert = TRUE]
+  write.report(weo, file = file.path(getConfig("outputfolder"), valfile), append = TRUE)
```

Comment:
So, question is: does `calcIEA_WEO_2021(subtype = 'region')` return different _global_ values then `calcIEA_WEO_2021(subtype = 'global')`, and if so: why?

Reply:
>  So, question is: does calcIEA_WEO_2021(subtype = 'region') return different global values then calcIEA_WEO_2021(subtype = 'global'), and if so: why?

This is all so complicated because we want to get as much as possible from the source.
There is global data available for more variables than regional data. And the regional data does not add up to global data.

So with subtype global, we take all the global data and disaggregate it with a very rough weight (because it is a formal madrat requirement), but we never want to use and data on sub-global level. 

For the subtype region, we look at the country and region level, which is available for less variables and even and the regions + countries dont add up to global. We take all we can get, but don't want to use the global value summed up from this, because it is incomplete. 

In the end you have a set with GLO from one calculation and some regional data from another calculation. Normally, in madrat, you either can disaggregate GLO, or read in countries and madrat aggregates to GLO + other regions.

Reply:
If not, then all variables from `calcIEA_WEO_2021(subtype = 'region')` can be filtered with `filter_historical_mif()`.

Reply:
> So, the strategy would be to modify convertIEA_WEO_2021() to disregard subtype, have it do what it is doing under subtype == 'region', but include the global figures as under subtype == 'global', and then filter for region/variable combinations we like.

This is not how convert functions work. Per convention, a convert function must return a magclass object with the 249 regions, otherwise it throws an error.

Reply:
So, it would be possible to 
- take the region/country values of the variables that are available,
- subtract them from the global totals,
- disaggregate the difference crudely across the countries for which there is no data (that way the aggregated sums would match the global values),
- and finally filter the result for global values and some regional values using `filter_historical_mif()`.

Not worth it.

To answer my question
> Why? There's a shiny new function to perform filtering like that? Why???

Because magclass.

Reply:
Yes, exactly. This would be possible, but very complicated.

Reply:
I don't think it's complicated.  Probably less complicated then what is done now.  But it would be a lot of work to invest.  Maybe earmark that for refactoring.